src/java/com/sarality/app/view/list/NonScrollingListViewInitializer.java

```diff
@@ -25,19 +25,30 @@
 public class NonScrollingListViewInitializer<T> extends BaseViewInitializer<LinearLayout, List<T>> {
 
   private final ListViewRowRenderer<T> rowRenderer;
+  private final ListViewRowRenderer<T> emptyViewRenderer;
 
   public NonScrollingListViewInitializer(FragmentActivity activity, LinearLayout listView,
-      ListViewRowRenderer<T> rowRenderer) {
+                                         ListViewRowRenderer<T> rowRenderer, ListViewRowRenderer<T> emptyViewRenderer) {
     super(activity, listView);
     this.rowRenderer = rowRenderer;
+    this.emptyViewRenderer = emptyViewRenderer;
   }
 
   @Override
   public void render(List<T> dataList) {
+    getView().removeAllViews();
+
+    if (dataList.size() == 0) {
+      LayoutInflater inflater = LayoutInflater.from(getContext());
+      View rowView = inflater.inflate(emptyViewRenderer.getRowLayout(null), getView());
+      emptyViewRenderer.render(rowView, null);
+    }
+
     for (T data : dataList) {
       LayoutInflater inflater = LayoutInflater.from(getContext());
-      View rowView = inflater.inflate(rowRenderer.getRowLayout(data), getView());
+      View rowView = inflater.inflate(rowRenderer.getRowLayout(data), getView(), false);
```

Comment:
the removeView/ addView and the false parameter to inflate was required because only the first rowView was getting added to the layout and the others were getting lost.